sys/net/gnrc/network_layer/sixlowpan/frag/rb/gnrc_sixlowpan_frag_rb.c

```diff
@@ -129,43 +129,68 @@ void gnrc_sixlowpan_frag_rb_add(gnrc_netif_hdr_t *netif_hdr,
     }
 }
 
+#ifndef NDEBUG
+static bool _valid_offset(gnrc_pktsnip_t *pkt, size_t offset)
+{
+    return (sixlowpan_frag_1_is(pkt->data) && (offset == 0)) ||
+           (sixlowpan_frag_n_is(pkt->data) &&
+            (offset == sixlowpan_frag_offset(pkt->data)));
+}
+#endif
+
+static uint8_t *_6lo_frag_payload(gnrc_pktsnip_t *pkt)
+{
+    if (sixlowpan_frag_1_is(pkt->data)) {
+        return ((uint8_t *)pkt->data) + sizeof(sixlowpan_frag_t);
+    }
+    else {
+        return ((uint8_t *)pkt->data) + sizeof(sixlowpan_frag_n_t);
+    }
+}
+
+static size_t _6lo_frag_size(gnrc_pktsnip_t *pkt, size_t offset, uint8_t *data)
+{
+    size_t frag_size;
+
+    if (offset == 0) {
+        frag_size = pkt->size - sizeof(sixlowpan_frag_t);
+        if (data[0] == SIXLOWPAN_UNCOMP) {
+            frag_size--;
```

Comment:
Please add a comment for people like me who are clueless about fragmentation - this one is not obvious.

Reply:
Because the "uncompressed datagram" dispatch is exactly one byte long and thus has to be removed from the fragment size. Will add a comment.

Reply:
(this code is just copied up from previously being in-line below before btw)